frontend/packages/ceph-storage-plugin/src/components/block-pool/body.tsx

```diff
@@ -0,0 +1,269 @@
+import { useK8sGet } from '@console/internal/components/utils/k8s-get-hook';
+import { useFlag } from '@console/shared/src/hooks/flag';
+
+import { CephClusterKind, StorageClusterKind } from '../../types';
+import { OCSServiceModel } from '../../models';
+import { CEPH_STORAGE_NAMESPACE, OCS_DEVICE_REPLICA } from '../../constants/index';
+import { checkArbiterCluster } from '../../utils/common';
+import { PROGRESS_STATUS } from '../../utils/storage-pool';
+import { POOL_STATE, POOL_PROGRESS } from '../../constants/storage-pool-const';
+import {
+  BlockPoolState,
+  BlockPoolAction,
+  BlockPoolActionType,
+  getErrorMessage,
+} from '../../utils/block-pool';
+import { GUARDED_FEATURES } from '../../features';
+
+import './body.scss';
+
+export const BlockPoolStatus: React.FC<BlockPoolStatusProps> = ({ status, name, error = '' }) => {
+  const { t } = useTranslation();
+  const statusObj = PROGRESS_STATUS(t).find((state) => state.name === status);
+
+  return (
+    <>
+      <EmptyState>
+        <EmptyStateIcon icon={statusObj.icon} className={statusObj.className} />
+        <EmptyStateBody data-test="empty-state-body">
+          {error ? getErrorMessage(error) : statusObj.desc.replace('{name}', name)}
+        </EmptyStateBody>
+      </EmptyState>
+    </>
+  );
+};
+
+export type BlockPoolStatusProps = {
+  status: string;
+  name?: string;
+  error?: string;
+};
+
+export const BlockPoolBody = (props: BlockPoolBodyPros) => {
+  const { cephCluster, state, dispatch, showPoolStatus } = props;
+  const { t } = useTranslation();
+
+  const isPoolManagementSupported = useFlag(GUARDED_FEATURES.OCS_POOL_MANAGEMENT);
+  const [storageCluster, storageClusterLoaded, storageClusterLoadError] = useK8sGet<
+    ListKind<StorageClusterKind>
+  >(OCSServiceModel, null, CEPH_STORAGE_NAMESPACE);
+
+  const [isReplicaOpen, setReplicaOpen] = React.useState(false);
+  const [isVolumeTypeOpen, setVolumeTypeOpen] = React.useState(false);
+  // Check storage cluster is in ready state
+  const isClusterReady: boolean = cephCluster?.status?.phase === POOL_STATE.READY;
+
+  // Check storage cluster is arbiter
+  React.useEffect(() => {
+    const isArbiterCluster: boolean = checkArbiterCluster(storageCluster?.items[0]);
+    dispatch({ type: BlockPoolActionType.SET_POOL_ARBITER, payload: isArbiterCluster });
+    if (isArbiterCluster) {
+      dispatch({ type: BlockPoolActionType.SET_POOL_REPLICA_SIZE, payload: '4' });
+    }
+  }, [
+    storageCluster,
+    storageClusterLoaded,
+    storageClusterLoadError,
+    state.isArbiterCluster,
+    dispatch,
+  ]);
+
+  React.useEffect(() => {
+    dispatch({
+      type: BlockPoolActionType.SET_ERROR_MESSAGE,
+      payload: !isClusterReady
+        ? t(
+            'ceph-storage-plugin~The OpenShift Storage cluster is not available. Try again after the storage cluster is ready to use.',
+          )
+        : '',
+    });
+  }, [dispatch, isClusterReady, t]);
```

Comment:
This error can be checked inside `onSubmitHandler` .
When you click create and then you render an error alert.
See add capacity modal error handling for storage class as a reference

Reply:
let me try this change